Source/JavaScriptCore/b3/B3Procedure.h

```diff
@@ -125,7 +125,7 @@ class Procedure {
     JS_EXPORT_PRIVATE const Vector<Type>& tupleForType(Type tuple) const;
 
     unsigned resultCount(Type type) const { return type.isTuple() ? tupleForType(type).size() : type.isNumeric(); }
-    Type typeAtOffset(Type type, unsigned index) const { ASSERT(index < resultCount(type)); return type.isTuple() ? extractFromTuple(type, index) : type; }
+    virtual Type typeAtOffset(Type type, unsigned index) const { ASSERT(index < resultCount(type)); return type.isTuple() ? extractFromTuple(type, index) : type; }
```

Comment:
`typeAtOffset` is not overridden anywhere, though, so this really shouldn't be virtual.